### PR TITLE
Added test for Cooperative Map & Cache Sources

### PR DESCRIPTION
**Pull request by olukas**
Added soak tests for sources which uses:
* local Map
* local Cache
* remote Map
* remote Cache
* querying in local Map
* querying in remote Map

Based on TS design, all of these tests are executed in one test case `CooperativeMapCacheSourceTest`.

### CODE REVIEW

File: cooperative-map-cache-source-test/src/main/java/com/hazelcast/jet/tests/cooperativesource/CooperativeMapCacheSourceTest.java [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+    }
+
+    private void initializeSourceMap() {
+        Map<Integer, String> map = jet.getMap(SOURCE_MAP);
+        for (int i = 0; i < SOURCE_MAP_ITEMS; i++) {
+            map.put(i, Integer.toString(i));
+        }
+    }
+
+    private ClientConfig wrappedRemoteClusterClientConfig() {
```
Comment (gurbuzali) [inline cooperative-map-cache-source-test/src/main/java/com/hazelcast/jet/tests/cooperativesource/CooperativeMapCacheSourceTest.java]: you can use `Util.uncheckCall`

File: cooperative-map-cache-source-test/src/main/java/com/hazelcast/jet/tests/cooperativesource/CooperativeMapCacheSourceTest.java [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+    private static final String TEST_PREFIX = CooperativeMapCacheSourceTest.class.getSimpleName();
+    private static final String SOURCE_MAP = TEST_PREFIX + "_SourceMap";
+    private static final String SINK_LOCAL_MAP = TEST_PREFIX + "_SinkLocalMap";
+    private static final String SINK_REMOTE_MAP = TEST_PREFIX + "_SinkRemoteMap";
+    private static final String SINK_LOCAL_CACHE = TEST_PREFIX + "_SinkLocalCache";
+    private static final String SINK_REMOTE_CACHE = TEST_PREFIX + "_SinkRemoteCache";
+    private static final String SINK_QUERY_LOCAL_MAP = TEST_PREFIX + "_SinkQueryLocalMap";
+    private static final String SINK_QUERY_REMOTE_MAP = TEST_PREFIX + "_SinkQueryRemoteMap";
+    private static final String SOURCE_CACHE = TEST_PREFIX + "_SourceCache";
+    private static final int SOURCE_MAP_ITEMS = 1_000_000;
```
Comment (gurbuzali) [inline cooperative-map-cache-source-test/src/main/java/com/hazelcast/jet/tests/cooperativesource/CooperativeMapCacheSourceTest.java]: I think 1M is too much data, we'll have 6M data in the cluster after the jobs complete

Comment (gurbuzali) [reply cooperative-map-cache-source-test/src/main/java/com/hazelcast/jet/tests/cooperativesource/CooperativeMapCacheSourceTest.java]: yes let's go with 100K and see how the cluster performs

**Comment by olukas** [comment]
Thanks for review @gurbuzali . I fixed your concerns.

**Pull request merged**